graphium/data/collate.py

```diff
@@ -86,14 +86,26 @@ def graphium_collate_fn(
         The batched elements. See `torch.utils.data.dataloader.default_collate`.
     """
 
+    # Skip any elements that failed
+    if None in elements:
+        elements = [e for e in elements if e is not None]
+
     elem = elements[0]
     if isinstance(elem, Mapping):
         batch = {}
         for key in elem:
             # Multitask setting: We have to pad the missing labels
             if key == "labels":
                 labels = [d[key] for d in elements]
-                batch[key] = collate_labels(labels, labels_size_dict, labels_dtype_dict)
+                if "features" in elem:
+                    num_nodes = [d["features"].num_nodes for d in elements]
+                    num_edges = [d["features"].num_edges for d in elements]
+                else:
+                    num_nodes = [d["num_nodes"] for d in elements]
+                    num_edges = [d["num_edges"] for d in elements]
```

Comment:
Done.  Because we know that the only possible keys in `MultitaskDataset` are now `"labels"`, `"features"`, `"num_nodes"`, and `"num_edges"`, is it safe to remove support for other keys and just access these 4 directly, instead of looping over the keys of one item?

Reply:
I think it's better to keep support for any key, in case a user wants to use other custom keys. Also, aren't some positional encodings using other keys?

Reply:
All feature data's under `"features"` and all label data's under `"labels"`.  `MultitaskDataset::__getitem__` does:

```
        if self.mol_file_data_offsets is None:
            datum = { "features": self.featurize_smiles(smiles_str) }
        else:
            datum = {
                "labels": self.load_graph_from_index(idx),
                "features": self.featurize_smiles(smiles_str),
            }
```

and returns `datum` after an error check.

Reply:
Ahh yeah, you're right. Still, I think collation should be flexible, and call back to `default_collate` for anything not `features` or `labels`